gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -2180,44 +2181,53 @@ void Movable::Backoff()
 void Movable::BumpAway()
 {
 	area->ClearSearchMapFor(this);
+	bumped = true;
 	area->AdjustPositionNavmap(Pos);
 }
 
 // Takes care of movement and actor bumping, i.e. gently pushing blocking actors out of the way
-// Returns whether the actor made all pending steps (so, false if it must be called again this tick).
 // The movement logic is a proportional regulator: the displacement/movement vector has a
 // fixed radius, based on actor walk speed, and its direction heads towards the next waypoint.
 // The bumping logic checks if there would be a collision if the actor was to move according to this
 // displacement vector and then, if that is the case, checks if that actor can be bumped
 // In that case, it bumps it and goes on with its step, otherwise it either stops and waits
 // for a random time (inspired by network media access control algorithms) or just stops if
 // the goal is close enough.
-bool Movable::DoStep(unsigned int walkScale, ieDword time) {
-	// Only bump back if you are not moving
+void Movable::DoStep(unsigned int walkScale, ieDword time) {
+	// Only bump back if not moving
+	// Actors can be bumped while moving if they are backing off
 	if (!path) {
-		if (OldPos == Pos) {
-			return true;
+		if (Destination != Pos) {
```

Comment:
Can we also reach Destination == Pos here? If so, wouldn't it be a case to cleanup and exit early?

Reply:
If `Destination == Pos` then either `IsBumped()` and you're bumping back (if oldPos is free) or `!IsBumped()` and you clean up and exit on the following if

Reply:
(If I got what you mean)